### PR TITLE
feat: email validation middleware added (#381)

### LINKED ISSUE #381: [ENHANCEMENT] Separate middleware for Email Validation
### What would you like to share?

In the userController.js file, the email validation is being carried out numerous times by defining it again and again in various POST requests. I believe it's a good practice to centralize validation logic to avoid duplication. I suggest defining a separate middleware like "validateEmail" and pass it onto the POST requests (or any other request) where the email validation needs to be carried out. Error handling can be modified similarly. 

I'd like to work on this enhancement issue myself. I am a beginner using Express, Node.Js and JS. This will be a great experience for me. So if you want, please assign this issue to me. Thanks :)

### Additional information

_No response_

### 👀 Have you checked if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Contributing Guidelines?

- [X] I have read and understood the rules in the [Contributing Guidelines](https://github.com/Dun-sin/Whisper/blob/main/CONTRIBUTING.md)

### Are you willing to work on this issue ?

Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by destryptor**
# Fixes Issue

**My PR closes #381**

# 👨‍💻 Changes proposed(What did you do ?)
I have added a custom middleware for email validation called "emailValidator" and have used it in the required request handlers. This aims to prevent duplication and is a modular approach to handle email-validation.

# ✔️ Check List (Check all the applicable boxes)

- [X] My code follows the code style of this project.
- [X] This PR does not contain plagiarized content.
- [X] The title and description of the PR is clear and explains the approach.

##  Note to reviewers

I believe the code should work fine just as before since I personally tested it, but if any issue arises, please let me know, I'll earnestly fix them.

# 📷 Screenshots

<a href="https://imgbb.com/">Screenshot-2023-10-06-155207</a>

### CODE REVIEW

**Comment by destryptor** [comment]
Please let me know if I did anything wrong, I have tried sticking to the rules strictly

**Comment by Dun-sin** [comment]
> Please let me know if I did anything wrong, I have tried sticking to the rules strictly

There's a new rule added in case you missed it, it's fine it was added today. It says to show a screen recording of your working software if you worked on the backend

File: server/controllers/userController.js [hunk 1]
```diff
@@ -3,125 +3,115 @@ const validator = require('validator').default;
...
+        if (!user) {
+            return res.status(404).json({ error: 'User not found' });
+        }
+
+        // Send the user profile data as JSON response
+        res.status(200).json(user);
+    } catch (error) {
+        console.error('Error fetching user profile:', error);
+        res.status(500).json({ error: 'Internal server error' });
+    }
```
Comment (Dun-sin) [inline server/controllers/userController.js:61]: remove the changes

**Review by Dun-sin — CHANGES_REQUESTED** [review]

**Comment by Dun-sin** [comment]
> > Please let me know if I did anything wrong, I have tried sticking to the rules strictly
> 
> There's a new rule added in case you missed it, it's fine it was added today. It says to show a screen recording of your working software if you worked on the backend

@destryptor

**Comment by destryptor** [comment]
> > > Please let me know if I did anything wrong, I have tried sticking to the rules strictly
> > 
> > 
> > There's a new rule added in case you missed it, it's fine it was added today. It says to show a screen recording of your working software if you worked on the backend
> 
> @destryptor

yeah just a second im uploading the recording on a cloud to embed it here

**Comment by destryptor** [comment]
https://github.com/Dun-sin/Whisper/assets/117581936/84f4a5f7-9e30-4445-8ed8-a46fd83db6c3


So what I've done to show functionality is that I've used Postman to send the POST and DELETE requests to the server where the emailValidator is required. I've sent JSONs with both valid and invalid emails to show that the validation works as intended.
Apart from that. the tab spaces have been fixed as you asked for.

**Review by Dun-sin — APPROVED** [review]
You did great, thanks for contributing, I hope you will stick around and continue to contribute to this project.

Consider giving this project a star and joining the community discord server if you haven't for more resources and opportunities to connect with others. [👉🏽here👈🏽](https://discord.com/invite/ufcysW9q23)

**Comment by destryptor** [comment]
Thanks! I've already starred the project. Thanks for giving me the opportunity:)

**Comment by Dun-sin** [comment]
> Thanks! I've already starred the project. Thanks for giving me the opportunity:)

you are welcome

**Pull request merged**